src/components/Header/Header.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { FC, memo, useState } from 'react';
+
+type Props = {
+  newTodoField: React.RefObject<HTMLInputElement>,
+  createTodo: (title: string) => void;
+  toggleAll: () => void;
+};
+
+export const Header: FC<Props> = memo((props) => {
+  const {
+    newTodoField,
+    createTodo,
+    toggleAll,
+  } = props;
+
+  const [title, setTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+
+  const onSubmit = () => {
+    setIsAdding(true);
+    createTodo(title);
+    setTitle('');
+    setIsAdding(false);
+  };
```

Comment:
Add validation here and try to clear the title only if todo was added successfully

Reply:
change everything for better way, like you show in the morning and now it has validation.